common/autoware_auto_perception_rviz_plugin/src/object_detection/predicted_objects_display.cpp

```diff
@@ -25,161 +28,239 @@ namespace object_detection
 {
 PredictedObjectsDisplay::PredictedObjectsDisplay() : ObjectPolygonDisplayBase("tracks")
 {
-  std::thread worker(&PredictedObjectsDisplay::workerThread, this);
-  worker.detach();
+  num_threads_property = new rviz_common::properties::IntProperty(
+    "Num Threads", 4, "Num of Threads to use", this);  // by default, use 4 threads
+  max_num_threads = 8;            // hard code the number of threads to be created
+  num_threads = max_num_threads;  // this will be soon modified in the workerThread
+
+  for (int ii = 0; ii < max_num_threads; ++ii) {
+    threads.emplace_back(std::thread(&PredictedObjectsDisplay::workerThread, this));
+  }
+
+  // ParallelizedThread Initialization
+  std::vector<std::thread> thread_vector;
+  ending_semaphores = reinterpret_cast<sem_t *>(malloc(max_num_threads * sizeof(sem_t)));
```

Comment:
Minor comment: the name `ending_semaphores` is not clear in that what "ending" means.